### DISCUSSION TITLE
List Tags ordered by timestamp using Github GraphQL

### DISCUSSION THREAD

**Discussion by raeeskkn** [community/community#37203] [Apps, API and Webhooks] [answered]
Here are the tags that were committed sequentially:
1.09.0 (on 24.09.2022)
1.11.0 (on 25.09.2022 07:00)
1.11.1 (on 25.09.2022 08:00)
1.10.0 (on 25.09.2022 09:00) ( some hotfix changes )

I would like to get the above tags in an array exactly as it is. For example 1.10.0 should be on the last index of array because 1.10.0 was submitted recently. However I am not succeeded.

I have listed all github Tags using GraphQL query:
`query 
  {repository(owner:"XXX", name:"XXX") 
    {refs(refPrefix: "refs/tags/", last: 100, orderBy: {field: TAG_COMMIT_DATE, direction: ASC}) {      
edges { 
        node {name}
        }
      }
    }
  }`
Here is the result:
... (array index ...)
1.09.0 (array index 96)
1.10.0 (array index 97)
1.11.0 (array index 98)
1.11.1 (array index 99)

How could I get 1.10.0 on the last array index (99) using graphql query? or how could I get github tags based on timestamp it was committed?

**Comment by LangLangBart** [comment, marked as answer]
I don't think it is supported by the `GraphQL API` or if it does it is well hidden. You would need another tool to get them listed.

### Local
Listing `tags`  locally would be simple:
```
git tag --sort='committerdate'
```

### GraphQL with GitHub's cli tool
[gh installation](https://github.com/cli/cli#installation)

#### QUERY
```graphql
# the query, which gives us the name of the tag and the date of the commit to which it refers
query ($owner: String!, $name: String!, $endCursor: String) {
  repository(owner: $owner, name: $name) {
    refs(refPrefix: "refs/tags/", first: 100, after: $endCursor) {
      totalCount
      pageInfo {
        endCursor
        hasNextPage
      }
      nodes {
        name
        target {
          ... on Commit {
            abbreviatedOid
            committedDate
          }
          ... on Tag {
            target {
              ... on Commit {
                abbreviatedOid
                committedDate
              }
            }
          }
        }
      }
    }
  }
}
```

#### Test
```zsh
# lets test it for stedolan/jq
gh api graphql --paginate --raw-field owner=stedolan --raw-field name=jq --raw-field query=$QUERY

# result
{
{
  "data": {
    "repository": {
      "refs": {
        "totalCount": 11,
        "pageInfo": {
          "endCursor": "MTE",
          "hasNextPage": false
        },
        "nodes": [
          {
            "name": "1.6rc2",
            "target": {
              "abbreviatedOid": "46d1ce2",
              "committedDate": "2018-08-18T03:23:10Z"
            }
          },
          {
            "name": "jq-1.0",
            "target": {
              "target": {
                "abbreviatedOid": "c53e001",
                "committedDate": "2012-10-21T22:26:31Z"
              }
            }
          },
....
```

#### Sort the result and return only the name of the tag as an array
`gh` has support for `jq` build-in, which allows you to specify which field to sort on
```zsh
gh api graphql --paginate --raw-field owner=stedolan --raw-field name=jq --raw-field query=$QUERY \
 --jq '.data.repository.refs.nodes |  [.[] | if .target.target? then {name:.name, date:.target.target.committedDate} else {name:.name, date:.target.committedDate} end] | sort_by(.date) | [.[].name]'

# result - https://github.com/stedolan/jq/tags
["jq-1.0","jq-1.1","jq-1.2","jq-1.3","jq-1.4","jq-1.5rc1","jq-1.5rc2","jq-1.5","jq-1.6rc1","1.6rc2","jq-1.6"]
```

![list_tags](https://user-images.githubusercontent.com/92653266/197921565-e5cc36be-39e8-42ad-b51e-e10e000a2fc1.gif)
